rfc/wip-tform/index.md

```diff
@@ -0,0 +1,679 @@
+- The `input` and `output` fields MUST be included for transformations of type: `mapAxis`, and `byDimension` (see the note
+    below), and under all other conditions.
+
+
+<dl>
+  <dt><strong>transformations</strong></dt>
+  <dd>A non-empty array of transformations.</dd>
+</dl>
+
+
+
+#### <a name=coordinates-displacements>coordinates and displacements</a>
+
+`coordinates` and `displacements` transformations store coordinates or displacements in an array and interpret them as a
+transformation. Applying the transformation amounts to looking up the appropriate locations in the array and interpolating
+if necessary.
+
+These transformation types refer to an array at location specified by the `"path"` parameter whose coordinate sytem MUST contain
+an axis with type `coordinate` or `displacement` respectively for transformations of type `coordinates` or `displacements`. This
+axis MUST be of length equal to the number of axes of the output coordinate system and SHOULD be the last axis (contiguous on
+disk when c-order). The coordinate system MUST also contain an axis identical to every axis of its input coordinate system in
+the same order. As a consequence, the array will have one dimension more than its input coordinate system.
+
+The `i`th value of the array along the `coordinate` or `displacement` axis refers to the displacement
+or coordinate of `i`th output axis. See the example below.
+
+
+`coordinates` and `displacements` transformations are not invertible in general, but implementations MAY approximate their
+inverses. Metadata for these coordinate transforms have the following field: 
+
+<dl>
+  <dt><strong>path</strong></dt>
+  <dd>  The location of the coordinate array in this (or another) container.</dd>
+  <dt><strong>interpolation</strong></dt>
+  <dd>  The `interpolation` attributes MAY be provided. It's value indicates
+        the interpolation to use if transforming points not on the array's discrete grid.
+        Values could be:
+        <ul>
+            <li><code>linear</code> (default)</li>
+            <li><code>nearest</code></li>
+            <li><code>cubic</code></li>
+        </ul></dd>
+</dl>
+
+For both `coordinates` and `displacements`, the array data at referred to by `path` MUST define space and coordinate transform metadata:
+
+* Every axis name in the `coordinateTransform`'s `input` MUST appear in the coordinate system
+* The array dimension corresponding to the `coordinate` or `displacement` axis MUST have length equal to the dimensionality of the `coordinateTransform` `output`
+* SHOULD have a `name` identical to the `name` of the corresponding `coordinateTransform`.
+
+For `coordinates`:
+
+* coordinateSystem metadata MUST have exactly one axis with `"type" : "coordinate"`
+* If a `coordinates`' input space has dimensionality `N`, then the array data at `path` MUST have dimensionality equal to `(N + 1)`.
+* the shape of the array along the "coordinate" axis must be exactly `N`
```

Comment:
I don't understand what this means: * the shape of the array along the "coordinate" axis must be exactly `N`

Reply:
Ah got it, I would maybe rephrase a bit, maybe say the "length of the array for the dimension of the `coordinate` axis". "shape` makes me think of `array.shape`, while here is `array.shape[-1] == n`.

Reply:
Yea, I'm referring to the shape field in the zarr's json (not python/numpy array shape). 
I could use "length" if it is clear enough that it refers to one entry of the shape...